### PR TITLE
[JENKINS-56172] Fix audit logging configuration output issue

### PR DESCRIPTION
**Pull request by davidolorundare**
See [JENKINS-56172](https://issues.jenkins-ci.org/browse/JENKINS-56172)

- Refactored the `log4j.xml` file to implement a fix for the audit-log config output issue.

### Desired Reviewers:
@jvz 
@jeffret-b

### CODE REVIEW

File: src/main/resources/log4j2.xml [hunk 1]
```diff
@@ -17,12 +17,11 @@
...
     <Properties>
-        <Property name="JENKINS_HOME">$${env:JENKINS_HOME:-.}</Property>
-        <Property name="FALLBACK_LOG_DESTINATION">$${JENKINS_HOME}/logs/audit.log</Property>
-        <Property name="AUDIT_LOG_DESTINATION">$${sys:auditFileName:-${FALLBACK_LOG_DESTINATION}}</Property>
+        <Property name="AUDIT_LOG_DESTINATION">${env:JENKINS_HOME:-.}/logs</Property>
+        <Property name="filename">${sys:auditFileName:-audit}</Property>
     </Properties>
     <Appenders>
-        <RollingRandomAccessFile name="audit" fileName="${AUDIT_LOG_DESTINATION}" filePattern="${AUDIT_LOG_DESTINATION}.%d{yyyyMMdd_HHmmss}-%i.log">
+        <RollingRandomAccessFile name="audit" fileName="${AUDIT_LOG_DESTINATION}/${filename}.log" filePattern="${AUDIT_LOG_DESTINATION}/archive/{filename}.log.%d{yyyyMMdd_HHmmss}-%i">
```
Comment (Lathaguna) [inline src/main/resources/log4j2.xml]: Hey,

What if I give an absolute file path for the logs to be stored in the UI?

**Comment by Lathaguna** [comment]
Hey,

I have doubt in thie fix. What if I give an absolute file path in the UI?

File: src/main/resources/log4j2.xml (hunk 1 above)
Comment (Lathaguna) [reply src/main/resources/log4j2.xml]:
```
<Properties>
        <Property name="DEFAULT_HOME">$${env:JENKINS_HOME:-.}</Property>
        <Property name="FALLBACK_LOG_DESTINATION">${DEFAULT_HOME}/logs/audit.log</Property>
        <Property name="AUDIT_LOG_DESTINATION">$${sys:auditFileName:-${FALLBACK_LOG_DESTINATION}}</Property>
    </Properties>
    <Appenders>
        <RollingRandomAccessFile name="audit" fileName="${AUDIT_LOG_DESTINATION}" filePattern="${AUDIT_LOG_DESTINATION}.%d{yyyyMMdd_HHmmss}-%i.log">
```

I had this change in my local for me to work. I am unable to submit a pr now unfortunately.

Comment (davidolorundare) [reply src/main/resources/log4j2.xml]: this fix was based on the assumption that the default location of the logs (as with the original `log42.xml` form before the task modified) was in the root directory where Jenkins is run. 

If the intent is to have the log folder be in other location outside of the root directory, then that's a rather different problem. Do you have an example of an absolute path that I can test with ?

Comment (jvz) [reply src/main/resources/log4j2.xml]: One example I could think of would be perhaps writing to a file in `/var/log/jenkins/` or some directory outside `$JENKINS_HOME`. Or perhaps there's a `/dev/foo` device you can set up that you can write logs to which collects logs from different programs. Or maybe writing to a socket file for some reason? Not sure which of these are completely possible, but something to think about.

**Comment by davidolorundare** [comment]
closing this PR since @Lathaguna has already submitted her fix.

**Pull request closed without merging**